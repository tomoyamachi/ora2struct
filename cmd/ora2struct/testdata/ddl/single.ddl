-- https://asktom.oracle.com/pls/apex/asktom.search?tag=ddl-export-from-database
CREATE TABLE "SCOTT"."EMP"
 (    "EMPNO" NUMBER(4,0),
      "ENAME" VARCHAR2(10),
      "JOB" VARCHAR2(9),
      "MGR" NUMBER(4,0),
      "HIREDATE" DATE,
      "HIREDATEN" DATE NOT NULL,
      TODATE DATE DEFAULT TO_DATE('99991231','YYYYMMDD') NOT NULL,
      "SAL" NUMBER(7,2),
      "COMM" NUMBER(7,2),
      "DEPTNO" NUMBER(2,0),
       CONSTRAINT "PK_EMP" PRIMARY KEY ("EMPNO")
USING INDEX PCTFREE 10 INITRANS 2 MAXTRANS 255
STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT)
TABLESPACE "SYSTEM"  ENABLE,
       CONSTRAINT "EMP_FK_EMP" FOREIGN KEY ("MGR")
        REFERENCES "SCOTT"."EMP" ("EMPNO") ENABLE,
       CONSTRAINT "FK_DEPTNO" FOREIGN KEY ("DEPTNO")
        REFERENCES "SCOTT"."DEPT" ("DEPTNO") ENABLE
 ) PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 NOCOMPRESS LOGGING
STORAGE(INITIAL 65536 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1 BUFFER_POOL DEFAULT);