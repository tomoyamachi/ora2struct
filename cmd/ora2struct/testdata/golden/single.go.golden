package models

import (
	"database/sql"
	"time"
)

// Table SCOTT.EMP
type Emp struct {
	Empno sql.NullInt64 `db:"EMPNO"` // type: NUMBER
	Ename sql.NullString `db:"ENAME"` // type: VARCHAR2
	Job sql.NullString `db:"JOB"` // type: VARCHAR2
	Mgr sql.NullInt64 `db:"MGR"` // type: NUMBER
	Hiredate sql.NullTime `db:"HIREDATE"` // type: DATE
	Hiredaten time.Time `db:"HIREDATEN"` // type: DATE
	Todate time.Time `db:"TODATE"` // type: DATE
	Sal sql.NullInt64 `db:"SAL"` // type: NUMBER
	Comm sql.NullInt64 `db:"COMM"` // type: NUMBER
	Deptno sql.NullInt64 `db:"DEPTNO"` // type: NUMBER
}

